packages/connections/src/modules/telemetry.spec.ts

```diff
@@ -0,0 +1,219 @@
+    });
+  },
+};
+
+describe('connection tracking', function () {
+  beforeEach(function () {
+    track = spy();
+  });
+
+  it('tracks a new connection attempt event - favorite', function () {
+    trackConnectionAttemptEvent(
+      { favorite: { name: 'example' }, lastUsed: null } as ConnectionInfo,
+      track
+    );
+    expect(track).to.have.been.calledWith('Connection Attempt', {
+      is_favorite: true,
+      is_recent: false,
+      is_new: true,
+    });
+  });
+
+  it('tracks a new connection attempt event - recent', function () {
+    trackConnectionAttemptEvent(
+      { favorite: undefined, lastUsed: new Date() } as ConnectionInfo,
+      track
+    );
+    expect(track).to.have.been.calledWith('Connection Attempt', {
+      is_favorite: false,
+      is_recent: true,
+      is_new: false,
+    });
+  });
+
+  it('tracks a new connection attempt event - new', function () {
+    trackConnectionAttemptEvent(
+      { favorite: undefined, lastUsed: undefined } as ConnectionInfo,
+      track
+    );
+    expect(track).to.have.been.calledWith('Connection Attempt', {
+      is_favorite: false,
+      is_recent: false,
+      is_new: true,
+    });
+  });
+
+  it('tracks a new connection attempt event - favorite and recent', function () {
+    trackConnectionAttemptEvent(
+      { favorite: { name: 'example' }, lastUsed: new Date() } as ConnectionInfo,
+      track
+    );
+    expect(track).to.have.been.calledWith('Connection Attempt', {
+      is_favorite: true,
+      is_recent: false,
+      is_new: false,
+    });
+  });
+
+  it('tracks a new connection event - localhost', async function () {
+    const connectionInfo = {
+      connectionOptions: {
+        connectionString: 'mongodb://localhost:27017',
+      },
+    } as ConnectionInfo;
+
+    trackNewConnectionEvent(connectionInfo, dataService, track);
+
+    expect(track).to.have.been.calledOnce;
+    const callback = track.getCall(0).args[1];
+    const response = await callback();
+    const expected = {
+      is_localhost: true,
+      is_public_cloud: false,
+      is_do_url: false,
+      is_atlas_url: false,
+      public_cloud_name: '',
+      auth_type: 'NONE',
+      tunnel: 'none',
+      is_srv: false,
+      is_atlas: false,
+      is_dataLake: false,
+      is_enterprise: false,
+      is_genuine: true,
+      non_genuine_server_name: 'na',
+      server_version: 'na',
+      server_arch: undefined,
+      server_os_family: undefined,
+    };
+
+    expect(expected).to.deep.equal(response);
+  });
+
+  it('tracks a new connection event - digital ocean', async function () {
+    const connectionInfo = {
+      connectionOptions: {
+        connectionString: 'mongodb://example.mongo.ondigitalocean.com:27017',
+      },
+    } as ConnectionInfo;
+
+    trackNewConnectionEvent(connectionInfo, dataService, track);
+
+    expect(track).to.have.been.calledOnce;
+    const callback = track.getCall(0).args[1];
+    const response = await callback();
```

Comment:
@mabaasit This is what this would look like without mocking:

```suggestion
    const trackEvent = once(process, 'compass:track');
    trackNewConnectionEvent(connectionInfo, dataService);

    const [ { properties: response } ] = await trackEvent;
```

with

```js
import { once } from 'events';
```

at the top of the file and

```js
(global as any)?.hadronApp = { isFeatureEnabled: () => true };
```

in the `before()` hook.

As you pointed out, this would probably be a bit nicer – we could get rid of the sinon mocking and the “fake” parameter to the `track*` functions, and this would also test the actual telemetry code, instead of mocking that out.